specification/sbi_atee.adoc

```diff
@@ -10,8 +39,1238 @@ shown in the figure above, the TEE Runtime (operating in VS-mode) supports
+
+[#table_sbi_tee_interrupt_set_tvm_aia_cpu_imsic_addr_errors]
+.TEE Interrupt Set TVM AIA CPU IMSIC Addr
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `tvm_vcpu_imsic_gpa` was invalid.
+| SBI_ERR_INVALID_PARAM   | `tvm_guest_id` or `tvm_vcpu_id` were invalid, or
+                            the TVM wasn't in the "initializing" state.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE Interrupt Convert AIA IMSIC (FID #2)
+[source, C]
+-------
+struct sbiret sbi_tee_interrupt_convert_aia_imsic(unsigned long imsic_page_addr);
+-------
+
+Starts the process of converting the non-confidential guest interrupt file at
+`imsic_page_addr` for use with a TVM. This must be followed by calls to `sbi_tee_host_global_fence()`
+and `sbi_tee_host_local_fence()` before the interrupt file can be assigned to a TVM.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_aia_tvm_convert_imsic_errors]
+.TEE Interrupt Convert AIA IMSIC
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `imsic_page_addr` was invalid.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE Interrupt Reclaim TVM AIA IMSIC (FID #3)
+[source, C]
+-------
+struct sbiret sbi_tee_interrupt_reclaim_tvm_aia_imsic(unsigned long imsic_page_addr);
+-------
+
+Reclaims the confidential TVM interrupt file at `imsic_page_addr`. The interrupt file
+must not currently be assigned to a TVM.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_reclaim_tvm_aia_imsic_errors]
+.TEE Interrupt Reclaim TVM AIA IMSIC
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `imsic_page_addr` was invalid.
+| SBI_ERR_INVALID_PARAM   | The memory is still assigned to a TVM.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+== TEE Guest Extension (EID 0x54454547)
+The TEE Guest extension supplements the TEE Host extension, and TVMs to communicate with TSM. A typical
+usecase for this extension is to relay information to the host.
+
+=== Function: TEE Guest Add Memory Region (FID #0)
+[source, C]
+-------
+struct sbiret sbi_tee_guest_add_memory_region(unsigned long tvm_memory_region_type,
+                                              unsigned long tvm_gpa_addr,
+                                              unsigned long region_len);
+-------
+Marks the range of TVM physical address space starting at `tvm_gpa_addr` as reserved for the mapping of
```

Comment:
update per dynamic memory conversion

Reply:
Changed...PTAL...we need to clarify some of the interactions with sbi_tee_guest_add_memory_region() and also MMIO...please see the discussion in the document.

Reply:
@rsahita Believe this is resolved...can you PTAL?

Reply:
`add_memory_region` has been replaced by `add_mmio_region`. Sharing/unsharing is handled with separate APIs.